src/model/block/payload/transaction/output/mod.rs

```diff
@@ -142,13 +144,26 @@ pub enum Output {
 
 impl Output {
     /// Returns the [`Address`] that is in control of the output.
-    pub fn owning_address(&self) -> Option<&Address> {
+    pub fn owning_address(&self, spent_timestamp: impl Into<Option<MilestoneTimestamp>>) -> Option<&Address> {
```

Comment:
Maybe I am misunderstanding something but isn't it the case that when the output is still unspent, but the latest milestone timestamp > expiration timestamp, then the owning address should already be the return address?

with the current logic the owner of an unspent output would still be the address in the address_unlock_condition despite we're passed the expiration.

so maybe this should just be MilestoneTimestamp and if the output is already spent, then it would be the spent_timestamp like currently, but when the output is still unspent then this should be the latest milestone timestamp.

Reply:
The problem is there's simply no good way for us to do this. We would have to update all unspent outputs every milestone, which is simply not performant. Spending proves definitively who the owner is, so we just update it then. Since this is only used in the explorer routes, this is fine. The only concern I have is with analytics, because we use it for active addresses, balance, and base token activity and those calculations may possibly be wrong now.